app/controllers/subscriptions_controller.rb

```diff
@@ -0,0 +1,24 @@
+require_relative "../models/mailchimp"
+
+class SubscriptionsController < ApplicationController
+  # Subscribe to the MailChimp newsletter.
+  def create
+    redirect_to request.referer if !current_user
```

Comment:
If CI can say what's wrong (whether that is test coverage or rubocop or failing tests or any of the above) that would be helpful.